pkg/osutil/interrupt_unix.go

```diff
@@ -69,6 +69,11 @@ func HandleInterrupts() {
 			os.Exit(0)
 		}
 		syscall.Kill(pid, sig.(syscall.Signal))
+
+		// unlock exit lock since the go runtime doesn't reinstall
```

Comment:
>  the golang runtime won't always reset the signal handler to default before sending the signal back to itself

can we somehow solve this instead? Or if etcd does not call exit, it will still hang there?